src/main/java/net/syscon/elite/service/MovementUpdateService.java

```diff
@@ -16,34 +22,56 @@
     private final ReferenceDomainService referenceDomainService;
     private final BedAssignmentHistoryService bedAssignmentHistoryService;
     private final BookingService bookingService;
+    private final OffenderBookingRepository offenderBookingRepository;
+    private final Clock clock;
 
-    public MovementUpdateService(ReferenceDomainService referenceDomainService, BedAssignmentHistoryService bedAssignmentHistoryService, BookingService bookingService) {
+    public MovementUpdateService(
+            final ReferenceDomainService referenceDomainService,
+            final BedAssignmentHistoryService bedAssignmentHistoryService,
+            final BookingService bookingService,
+            final OffenderBookingRepository offenderBookingRepository,
+            final Clock clock) {
         this.referenceDomainService = referenceDomainService;
         this.bedAssignmentHistoryService = bedAssignmentHistoryService;
         this.bookingService = bookingService;
+        this.offenderBookingRepository = offenderBookingRepository;
+        this.clock = clock;
     }
 
-    // @VerifyBookingAccess TODO DT-235 put this back - make sure it has a test dedicated to it
-    public OffenderSummary moveToCell(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime dateTime) {
+    @Transactional
+    @VerifyBookingAccess
+    @HasWriteScope
+    public OffenderBooking moveToCell(final Long bookingId, final Long livingUnitId, final String reasonCode, final LocalDateTime dateTime) {
+        validateMoveToCell(reasonCode, dateTime);
+        final var movementDateTime = dateTime != null ? dateTime : LocalDateTime.now(clock);
         referenceDomainService.getReferenceCodeByDomainAndCode(CELL_MOVE_REASON.getDomain(), reasonCode, false);
-        final var offenderSummary = getOffenderSummary(bookingId);
+        final var offenderBooking = getActiveOffenderBooking(bookingId);
 
-        if (offenderSummary.getInternalLocationId().equals(String.valueOf(livingUnitId))) {
-            return offenderSummary;
+        if (offenderBooking.getAssignedLivingUnitId().equals(livingUnitId)) {
+            return offenderBooking;
         }
 
-// TODO DT-235 Uncomment the updates - this is currently still a work in progress and we don't want to actually update anything yet
-//        bookingService.updateLivingUnit(bookingId, livingUnitId);
-//        bedAssignmentHistoryService.add(bookingId, livingUnitId, reasonCode, dateTime);
-        return getOffenderSummary(bookingId);
+        bookingService.updateLivingUnit(bookingId, livingUnitId);
```

Comment:
BTW - if you didn't do this update of booking does the trigger update anyway?

Reply:
No. All the trigger does is update the end dates on the previous bed assignment history record.  I suspect all of the other procedures are called from the Oracle forms...

Reply:
In fact this method performs the same updates as the process_release_bed procedure - update living unit and add to bed assignment histories.